components/calendar/src/astronomy.rs

```diff
@@ -1832,9 +1840,14 @@ impl Astronomical {
     /// Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/9afc1f3/calendar.l#L4132-L4146
     pub(crate) fn estimate_prior_solar_longitude(angle: f64, moment: Moment) -> Moment {
         let rate = MEAN_TROPICAL_YEAR / 360.0;
-        let tau =
-            moment - rate * div_rem_euclid_f64(Self::solar_longitude(moment) - angle, 360.0).1;
-        let delta = interval_mod_f64(Self::solar_longitude(tau) - angle, -180.0, 180.0);
+        let julian_centuries = Self::julian_centuries(moment);
+        let tau = moment
+            - rate * div_rem_euclid_f64(Self::solar_longitude(julian_centuries) - angle, 360.0).1;
+        let delta = interval_mod_f64(
+            Self::solar_longitude(Self::julian_centuries(tau)) - angle,
```

Comment:
Praise: good job not just passing in `julian_centuries` here.

Reply:
thanks :)